base/osutils.jl

```diff
@@ -1,20 +1,26 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-"""
-    OS_NAME
+let UNAME = ccall(:jl_get_UNAME, Any, ())
+    # change some `uname -s` values to "friendly" names
+    # for internal usage
```

Comment:
> If we built in cygwin the kernel would still be NT

the cygwin kernel is cygwin. sure, it's implemented in userspace, but it isn't really compatible with `nt`

> Same in WSL

Initial reports seem indicate that Microsoft's kernel is going to lie and call itself `Linux`. This is highly unfortunately. It should have been given it's own name so that we could write hybrid applications against it like we do on all other *nix platforms.

>  I don't think we expose any linux kernel syscalls at the Julia level either, we either call into posix libc or call public libuv functions which generally go by the libc name rather than the kernel specific details.

The names have common inspirations in many cases from the posix API (the libc names and the kernel names are also often nearly similar). But using `libuv` as our compatibility layer allows us to ignore limitations in the userspace libc and directly call into the kernel (it implements the syscalls directly for most newer API's).

> We aren't calling NT syscalls, we're calling the win32 os api

We're stuck with the Win32 compatibility layer in the short term. I'm still hoping that someday Microsoft will release a more functional interface to the NT kernel (such as `midipix`) that we can transition to instead.

> uname is most often executed from shell scripts

I would even limit that more to say it is most often executed by shell build configuration scripts. And as it happens, that's also exactly the same reason it's being called here (using Julia as the shell program and with a build output of sys.so).